vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java

```diff
@@ -395,25 +398,77 @@ public void updateSelection(Set<TItem> addedItems,
     }
 
     /**
-     * Gets whether displaying of all selected item labels as chips is enabled.
+     * Defines possible behavior of the component when not all selected items
+     * can be displayed as chips within the current field width.
+     */
+    public enum AutoExpandMode {
+        /**
+         * Field expands vertically and chips wrap.
+         */
+        VERTICAL,
+
+        /**
+         * Field expands horizontally until max-width is reached, then collapses
+         * to overflow chip.
+         */
+        HORIZONTAL,
+
+        /**
+         * Field expands horizontally until max-width is reached, then expands
+         * vertically and chips wrap.
+         */
+        BOTH,
+
+        /**
+         * Field does not expand and collapses to overflow chip.
+         */
+        NONE,
+    }
+
+    /**
+     * Gets the behavior of the component when not all selected items can be
+     * displayed as chips within the current field width.
      *
      * @since 24.3
-     * @return {@code true} if enabled, {@code false} otherwise
+     * @return {AutoExpandMode}
      */
-    public boolean isAllChipsVisible() {
-        return getElement().getProperty("allChipsVisible", false);
+    public AutoExpandMode getAutoExpand() {
+        return autoExpand;
     }
 
     /**
-     * Enables or disables displaying of all selected item labels as chips. When
-     * set to {@code false}, chips that do not fit are collapsed.
+     * Sets the behavior of the component when not all selected items can be
+     * displayed as chips within the current field width.
      *
+     * Expansion only works with undefined size in the desired direction (i.e.
+     * setting `max-width` limits the component's width).
+     *
+     * @param {AutoExpandMode}
+     *            autoExpandMode
      * @since 24.3
-     * @param autoOpen
-     *            {@code true} to always show all chips
      */
-    public void setAllChipsVisible(boolean allChipsVisible) {
-        getElement().setProperty("allChipsVisible", allChipsVisible);
+    public void setAutoExpand(AutoExpandMode autoExpandMode) {
+        autoExpand = autoExpandMode;
```

Comment:
What is the expected behaviour if I pass `null`as mode? Does it reset the properties or fail with an exception? 

nit / styling: I would personally change the enum to take 2 parameter for `autoExpandHorizontally` and `autoExpandVertically` so the whole method could be reduced to getElement().set("a", autoExpandMode.getA()) + B.

Reply:
> What is the expected behaviour if I pass nullas mode? Does it reset the properties or fail with an exception?

Good catch, thanks. I think it should fail.

Reply:
> nit / styling: I would personally change the enum to take 2 parameter for autoExpandHorizontally and autoExpandVertically so the whole method could be reduced to getElement().set("a", autoExpandMode.getA()) + B.

I'd also recommend this refactor as it cleans up the setter quite a bit

Reply:
> I'd also recommend this refactor as it cleans up the setter quite a bit

Thanks, done.